src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -363,6 +395,44 @@ protected function getIdFromValue(string $value)
     }
 
     /**
+     * Transforms IRI to raw identifiers if possible.
+     *
+     * @return array
+     */
+    protected function getValueIdentifiers(array $values, string $field)
+    {
+        foreach ($values as $key => $value) {
+            if (is_numeric($value)) {
+                continue;
+            }
+
+            try {
+                $identifiers = $this->iriConverter->getIdentifiersFromIri($value, ['fetch_data' => false]);
+                $values[$key] = $identifiers[$field] ?? reset($identifiers); //an assumption is being made for composite identifiers
+            } catch (InvalidArgumentException $e) {
+                // Do nothing, return the raw value
+            }
+        }
+
+        return $values;
+    }
+
+    /**
+     * When the field should be an integer, check that the given value is a valid one.
+     */
+    protected function hasValidValues(array $values, string $field, ClassMetadata $metadata): bool
+    {
+        foreach ($values as $key => $value) {
+            $type = $metadata->getTypeOfField($field);
+            if (DBALType::INTEGER === $type && null !== $value && false === filter_var($value, FILTER_VALIDATE_INT)) {
+                return false;
```

Comment:
Shouldn't we throw a `400 Bad Request` error instead of silently ignoring such errors (I'm not sure, just asking).

Reply:
I asked myself the same but as we're ignoring silently bad properties in other filters (just logging) I kept the same behavior here. However, this is a BC break as [this test](https://github.com/api-platform/core/pull/1480/files#diff-534f9c66d94968377ac76a8bc211da4dR386) is now getting results.